### PR TITLE
Module hierarchy

### PR DESCRIPTION
**Pull request by simonleary-umass-edu**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -2,86 +2,70 @@
...
-Need additional help?  We offer office hours every week on Tuesdays 2-4 PM on [Zoom](https://umass-amherst.zoom.us/j/95663998309?pwd=K2F2b0ZrNmhYR3pic1loY2pvcUhkdz09). Be sure to check the [cluster notes page](https://unity.rc.umass.edu/index.php) for up-to-date information on any canceled/delayed office hours.
+Need additional help?  We offer office hours every week on Tuesdays 2:30-4 PM on [Zoom](https://umass-amherst.zoom.us/j/95663998309?pwd=K2F2b0ZrNmhYR3pic1loY2pvcUhkdz09). Be sure to check the [cluster notes page](https://unity.rc.umass.edu/index.php) for up-to-date information on any canceled/delayed office hours.
 
-**Need expert help** using the cluster and optimizing your code?  We encourage you to schedule an appointment with one of our experienced HPC facilitators.  Send an email to <hpc@umass.edu> and request an HPC facilitator consultation.
+**Need expert help?** using the cluster and optimizing your code?  We encourage you to schedule an appointment with one of our experienced HPC facilitators.  Send an email to <hpc@umass.edu> and request an HPC facilitator consultation.
```
Comment (bryank-cs) [inline docs/index.md:20]: This adds a second question mark in the middle of the sentence.

File: docs/index.md [hunk 2]
```diff
@@ -2,86 +2,70 @@
...
+[SSH](connecting/ssh.md) is the more traditional method of using an HPC cluster. You will connect to the login node of unity, and you will be responsible for starting your own jobs. This can be more useful than JupyterHubs for jobs that last a long time and must be left unattended, or to have much more refined control over the resources allocated for your job.
 
-#### 3. Requesting Resources ####
-If you are on an SSH connection, you will have to manually request resources. Once you decide on what resources you want, you will submit that information to the scheduler, which will place you in a queue. If your resources are available immediately, your request will return right away, if not, you will be held in the queue until your requested resources become available.
+## Requesting Resources ##
+If you are on an SSH connection, you will have to manually request resources. Once you decide on what resources you want, you will submit that information to the scheduler, which will place you in a partition. If your resources are available immediately, your request will return right away, if not, you will be held in the partition until your requested resources become available.
```
Comment (bryank-cs) [inline docs/index.md:37]: "held in the partition" doesn't sound quite right. Maybe somewhere before this mention that partitions operate as queues and leave this usage as queue. Also "place *you* in a partition" doesn't work. The job is placed there, not the individual. Finally, "you request will return right away" doesn't make sense for a long running batch job. I think this was intended to describe an interactive session, but that's not clear from the context.

File: docs/index.md [hunk 3]
```diff
@@ -2,86 +2,70 @@
...
-4. Once your job completes, the result returns to the client.
\ No newline at end of file
+1. Client connects to Unity using [SSH](connecting/ssh.md) or [JupyterHub](connecting/jupyter.md).
+1. Client schedules a job through Slurm.
+1. Slurm allocates resources and connects the client to them.
+1. Job runs.
+    * Interactive job: Client does their work until they close the session or run out of time.
+    * Batch job: The job runs until it finishes its tasks or runs out of time.
+1. Job returns output.
+    * Batch job: The output to the command line is saved by slurm to a file like `slurm-xyz.out`.
```
Comment (bryank-cs) [inline docs/index.md:71]: Better might be: "The console output of the commands in the batch script are written to a specified file, or `slurm-###.out`."

File: docs/faq.md [hunk 4]
```diff
@@ -1,36 +1,34 @@
...
 
-### Can I run containers on Unity? 
+### Can I run containers on Unity?
 Yes! We support singularity containers, which are fully compatible with docker images. Run "module load singularity" to access it.
 
-### How much storage do I get on Unity and is it backed up? 
+### How much storage do I get on Unity and is it backed up?
 Refer to storage information [here](https://unity.rc.umass.edu/docs/technical/storage.html).
 We do not provide backup solutions by default.
 We take a snapshot of all storage at 1 AM every day for the past 48 hours.
```
Comment (drparker2022) [inline docs/faq.md:34]: We do not take snapshots of NESE storage. I don't know what, if anything, is done with group-specific storage such as ceewater-nas[12], astroth-nas and zhoulin-nas. This should probably just say we take a snapshot of /home and /work directories.

File: docs/index.md [hunk 5]
```diff
@@ -2,86 +2,70 @@
...
+Need additional help?  We offer office hours every week on Tuesdays 2:30-4 PM on [Zoom](https://umass-amherst.zoom.us/j/95663998309?pwd=K2F2b0ZrNmhYR3pic1loY2pvcUhkdz09). Be sure to check the [cluster notes page](https://unity.rc.umass.edu/index.php) for up-to-date information on any canceled/delayed office hours.
 
-**Need expert help** using the cluster and optimizing your code?  We encourage you to schedule an appointment with one of our experienced HPC facilitators.  Send an email to <hpc@umass.edu> and request an HPC facilitator consultation.
+**Need expert help?** using the cluster and optimizing your code?  We encourage you to schedule an appointment with one of our experienced HPC facilitators.  Send an email to <hpc@umass.edu> and request an HPC facilitator consultation.
 
 
+## Connecting to the cluster ##
+You can connect to Unity in two ways, an SSH connection (the standard linux console), or an instance of JupyterHub:
```
Comment (drparker2022) [inline docs/index.md:24]: linux --> Linux

File: docs/index.md (hunk 3 above)
Comment (drparker2022) [reply docs/index.md:71]: I would go with slurm-<job ID>.out to help the user identify where the job's output goes (especially after a few of these accumulate) and realize it's not a random number.

File: docs/slurm/mpi.md [hunk 6]
```diff
@@ -9,13 +9,13 @@
 #### srun ####
 ```
 module load openmpi
-mpicc /modules/admin-resources/mpi_testing/mpi_array.c -o mpi_array
+mpicc /software/admin-resources/mpi_testing/mpi_array.c -o mpi_array
```
Comment (drparker2022) [inline docs/slurm/mpi.md:12]: Am I missing something? Where is /software?

File: docs/slurm/mpi.md [hunk 7]
```diff
@@ -9,13 +9,13 @@
...
 srun --pty -N 2 mpi_array
 ```
 #### sbatch ####
 ```
 module load openmpi
-mpicc /modules/admin-resources/mpi_testing/mpi_array.c -o mpi_array
+mpicc /software/admin-resources/mpi_testing/mpi_array.c -o mpi_array
```
Comment (drparker2022) [inline docs/slurm/mpi.md:18]: Am I missing something? Where is /software?

**Comment by simonleary-umass-edu** [comment]
`/software` was a find and replace error, it's gone now

File: docs/software/conda.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+conda activate ./testName
+```
+
+Your currently active conda environment will appear in parentheses to the left of your command line prompt:
+```
+user@login2:~$
+conda activate ./testName
+(testName) user@login2:~$
+```
+
```
Comment (bryank-cs) [inline docs/software/conda.md:66]: Lines 62 and 63 should be combined (the command should be next to the prompt).

**Review by bryank-cs — APPROVED** [review]

**Pull request merged**